stl/inc/format

```diff
@@ -666,6 +666,10 @@ public:
         : _Active_state(_Basic_format_arg_type::_Int_type), _Int_state(_Val) {}
     explicit basic_format_arg(const unsigned int _Val) noexcept
         : _Active_state(_Basic_format_arg_type::_UInt_type), _UInt_state(_Val) {}
+    explicit basic_format_arg(const long _Val) noexcept
+        : _Active_state(_Basic_format_arg_type::_Int_type), _Int_state(_Val) {}
+    explicit basic_format_arg(const unsigned long _Val) noexcept
+        : _Active_state(_Basic_format_arg_type::_UInt_type), _UInt_state(_Val) {}

```

Comment:
I would prefer to just static cast to the storage type when creating the basic_format_arg, instead of messing around with this constructor. This overload set isn't really as horrible as the real mapping done by the phony overload set and storage type but still.

Actually: We probably shouldn't have this overload set at all, and instead just have an initializer from memory and a type enumerator, along with the T&& constructor that does the cast.

This would consolidate this overload set and the one in basic_format_args::get

This would also mean the basic_format_arg constructor could actually behave as the standard says it should (even though it's exposition only, so doesn't have to.

Reply:
Actually, I retract my suggestion. If we add the overloads we don't need to mess around with the formatter when we consolidate this overload set with the phony one.